File: uikit/src/main/java/utils/KeyBoardUtils.java
package utils;

import android.app.Activity;
import android.view.View;
import android.view.ViewTreeObserver;

public class KeyBoardUtils {

    static int mAppHeight;

    static int currentOrientation = -1;

    public static void setKeyboardVisibilityListener(final Activity activity, final View contentView, final KeyboardVisibilityListener keyboardVisibilityListener) {

        contentView.getViewTreeObserver().addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {

            private int mPreviousHeight;

            @Override

            public void onGlobalLayout() {

                int newHeight = contentView.getHeight();

                if (newHeight == mPreviousHeight)

                    return;

                mPreviousHeight = newHeight;

                if (activity.getResources().getConfiguration().orientation != currentOrientation) {

                    currentOrientation = activity.getResources().getConfiguration().orientation;

                    mAppHeight = 0;

                }

                if (newHeight >= mAppHeight) {

                    mAppHeight = newHeight;

                }

                if (newHeight != 0) {

                    if (mAppHeight > newHeight) {


                        keyboardVisibilityListener.onKeyboardVisibilityChanged(true);

                    } else {


                        keyboardVisibilityListener.onKeyboardVisibilityChanged(false);

                    }

                }

            }

        });

    }
}

File: uikit-kotlin/src/main/java/com/cometchat/pro/uikit/ui_resources/utils/keyboard_utils/KeyBoardUtils.kt
package com.cometchat.pro.uikit.ui_resources.utils.keyboard_utils

import android.app.Activity
import android.view.View
import android.view.ViewTreeObserver.OnGlobalLayoutListener

public class KeyBoardUtils {
    companion object {
        var mAppHeight = 0

        var currentOrientation = -1

        fun setKeyboardVisibilityListener(activity: Activity, contentView: View, keyboardVisibilityListener: KeyboardVisibilityListener) {
            contentView.viewTreeObserver.addOnGlobalLayoutListener(object : OnGlobalLayoutListener {
                private var mPreviousHeight = 0
                override fun onGlobalLayout() {
                    val newHeight = contentView.height
                    if (newHeight == mPreviousHeight) return
                    mPreviousHeight = newHeight
                    if (activity.resources.configuration.orientation != currentOrientation) {
                        currentOrientation = activity.resources.configuration.orientation
                        mAppHeight = 0
                    }
                    if (newHeight >= mAppHeight) {
                        mAppHeight = newHeight
                    }
                    if (newHeight != 0) {
                        if (mAppHeight > newHeight) {
                            keyboardVisibilityListener.onKeyboardVisibilityChanged(true)
                        } else {
                            keyboardVisibilityListener.onKeyboardVisibilityChanged(false)
                        }
                    }
                }
            })
        }
    }
}